WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -87,19 +87,41 @@ def index
     end
 
     if @display == 'admin'
-      @competitions = @competitions.includes(:delegates, :delegate_report)
+      @competitions = @competitions.includes(:delegate_report)
     end
 
     unless params[:region] == "all"
-      @competitions = @competitions.select { |competition| competition.belongs_to_region?(params[:region]) }
+      @competitions = @competitions.where("countryId = ? or continentId = ?", params[:region], params[:region])
     end
 
     if params[:search].present?
-      @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
+      params[:search].split.all? do |part|
+        @competitions = @competitions.where(
+          "Competitions.name like ? or
+          Competitions.cityName like ? or
+          users.name like ? or
+          Competitions.venue like ? or
+          Competitions.cellName like ? or
+          Competitions.countryId like ?",
+          "%#{part}%",
+          "%#{part}%",
+          "%#{part}%",
+          "%#{part}%",
+          "%#{part}%",
+          "%#{part}%",
```

Comment:
I agree with @jonatanklosko on both parts. It is hard to see those `%`, but they're a key part of providing this nice, expensive search feature =)